### PR TITLE
feat(dialog): describe dialog and dialog full screen with design tokens

### PR DESCRIPTION
**Pull request by mjaniczko-sage**
### Proposed behaviour
Describes the Dialog and DialogFullScreen with design tokens.

### Current behaviour
The Dialog and DialogFullScreen use the theme styled-component property.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by jhajduk-sage — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

File: src/components/dialog-full-screen/dialog-full-screen.style.js [hunk 1]
```diff
@@ -11,13 +11,15 @@ import {
...
+  background-color: var(--colorsUtilityMajor025);
   height: 100%;
   left: 0;
   position: fixed;
   top: 0;
   width: 100%;
-  z-index: ${({ theme }) => theme.zIndex.fullScreenModal};
+  z-index: ${({ theme }) =>
+    theme.zIndex
+      .fullScreenModal}; // TODO (tokens): implement elevation tokens - FE-4437
```
Comment (Parsium) [inline src/components/dialog-full-screen/dialog-full-screen.style.js:22]: Personally I think we should be adding these kind of reminders in the ticket description on our backlog, so they're all in one place and aren't public-facing

**Review by Parsium — CHANGES_REQUESTED** [review]

File: src/components/dialog-full-screen/dialog-full-screen.style.js (hunk 1 above)
Comment (Parsium) [reply src/components/dialog-full-screen/dialog-full-screen.style.js:22]: I've added to the ticket description this file needs changing :)

Comment (mjaniczko-sage) [reply src/components/dialog-full-screen/dialog-full-screen.style.js:22]: There is already few comments like that in code base. We can delete them all if needed. There's no need to add files to task - we can search through all code base

Comment (Parsium) [reply src/components/dialog-full-screen/dialog-full-screen.style.js:22]: This is only a minor thing, so if several comments exist currently then that should be fine. Personally though going forward, I think we should be placing any written reminders for future work along with the respective ticket.

**Review by Parsium — APPROVED** [review]

**Review by DianeGit22 — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Review by DianeGit22 — APPROVED** [review]

**Pull request merged**